client/src/controllers/TabsController.ts

```diff
@@ -0,0 +1,280 @@
+import { Controller } from '@hotwired/stimulus';
+
+const TAB_DEFAULTS = {
+  css: {
+    animate: 'animate-in',
+  },
+  keys: {
+    end: 'End',
+    home: 'Home',
+    left: 'ArrowLeft',
+    up: 'ArrowUp',
+    right: 'ArrowRight',
+    down: 'ArrowDown',
+  },
+  direction: {
+    ArrowLeft: -1,
+    ArrowRight: 1,
+  },
+}
+
+export class TabsController extends Controller<HTMLDivElement> {
+  static targets = ['tabList', 'tabTrigger', 'tabLabel', 'tabPanel'];
+
+  static values = {
+    transition: { default: 150, type: Number },
+    activeTabId: { default: '', type: String },
+    disableURL: { default: false, type: Boolean },
+    initialPageLoad: { default: true, type: Boolean },
+    animate: { default: true, type: Boolean },
+  };
+
+  declare tabListTarget: HTMLDivElement;
```

Comment:
```suggestion
  declare tabListTarget: HTMLElement;
```

No need for being to specific on the div here.